pkg/util/validation/exporter/exporter.go

```diff
@@ -84,3 +144,83 @@ func (oe *OverridesExporter) Collect(ch chan<- prometheus.Metric) {
 		ch <- prometheus.MustNewConstMetric(oe.overrideDescription, prometheus.GaugeValue, float64(limits.RulerMaxRuleGroupsPerTenant), "ruler_max_rule_groups_per_tenant", tenant)
 	}
 }
+
+// RingHandler is an http.Handler that serves requests for the overrides-exporter ring status page
+func (oe *OverridesExporter) RingHandler(w http.ResponseWriter, req *http.Request) {
+	if oe.ring != nil {
+		oe.ring.lifecycler.ServeHTTP(w, req)
+		return
+	}
+
+	ringDisabledPage := `
+		<!DOCTYPE html>
+		<html>
+			<head>
+				<meta charset="UTF-8">
+				<title>Overrides-exporter Status</title>
+			</head>
+			<body>
+				<h1>Overrides-exporter Status</h1>
+				<p>Overrides-exporter hash ring is disabled.</p>
+			</body>
+		</html>`
+	util.WriteHTMLResponse(w, ringDisabledPage)
+}
+
+// ownsTenant determines whether this overrides-exporter instance owns the given
+// tenant. If the ring is disabled, it assumes this instance has ownership of all
+// tenants. If the ring is enabled, it uses it to check ownership of the given
+// tenant.
+func (oe *OverridesExporter) ownsTenant(tenantID string) bool {
+	if oe.ring == nil {
+		// If sharding is not enabled, every instance exports metrics for every tenant
+		return true
+	}
+	owned, err := oe.ring.Owns(tenantID)
+	if err != nil {
+		_ = level.Warn(oe.logger).Log("msg", "determining tenant ownership failed", "err", err.Error())
```

Comment:
[nit] No need to assign the returned error to `_`. The linter has a custom rule to exclude `Log()` returned error from its checks. Also, please clarify it's the overrides-exporter, otherwise from the log would be hard to understand who is having this issue.

```suggestion
		level.Warn(oe.logger).Log("msg", "overrides-exported failed to determine tenant ownership", "err", err.Error())
```